src/com/amazon/ion/impl/bin/utf8/Pool.java

```diff
@@ -1,6 +1,8 @@
 package com.amazon.ion.impl.bin.utf8;
```

Comment:
This class started out as a `Utf8EncoderPool` and was generalized when we realized how many different resources we could be pooling. We should consider promoting `Pool<T>` to a higher-level package.

Reply:
True; I'll leave that for its own PR. Issue to track: https://github.com/amzn/ion-java/issues/406